src/mixins/object_geometry.mixin.js

```diff
@@ -572,14 +572,14 @@
     },
 
     calcOCoords: function() {
-      var rotateMatrix = this._calcRotateMatrix(),
-          translateMatrix = this._calcTranslateMatrix(),
-          vpt = this.getViewportTransform(),
-          startMatrix = this.group ? multiplyMatrices(vpt, this.group.calcTransformMatrix()) : vpt,
-          startMatrix = multiplyMatrices(startMatrix, translateMatrix),
-          finalMatrix = multiplyMatrices(startMatrix, rotateMatrix),
-          finalMatrix = multiplyMatrices(finalMatrix, [1 / vpt[0], 0, 0, 1 / vpt[3], 0, 0]),
-          dim = this._calculateCurrentDimensions(),
+      var vpt = this.getViewportTransform(),
+          tMatrix = this._calcTranslateMatrix(true),
+          rMatrix = this._calcRotateMatrix(true, !!this.group),
+          positionMatrix = multiplyMatrices(tMatrix, rMatrix),
+          startMatrix = multiplyMatrices(vpt, positionMatrix),
+          finalMatrix = multiplyMatrices(startMatrix, [1 / vpt[0], 0, 0, 1 / vpt[3], 0, 0]),
+          transformOptions = this.group ? fabric.util.qrDecompose(this.calcTransformMatrix()) : undefined,
+          dim = this._calculateCurrentDimensions(transformOptions),
```

Comment:
since the changes we are doing to rotate matrix and translate matrix are used here only, i wouldn't extend the api.
Let's handle the difference in this function.
Something like:
```js
    var vpt = this.getViewportTransform(),
          center = this.getCenterPoint(),
          angle = this.getTotalAngle(),
          finalAngle = !!this.group && this.flipX ? angle - 180 : angle;
          rMatrix = util.calcRotateMatrix({ angle: finalAngle });
    rMatrix[4] = center.x; rMatrix[5] = center.y;
    var startMatrix = multiplyMatrices(vpt, rMatrix),
          finalMatrix = multiplyMatrices(startMatrix, [1 / vpt[0], 0, 0, 1 / vpt[3], 0, 0]),
          transformOptions = this.group ? fabric.util.qrDecompose(this.calcTransformMatrix()) : undefined,
          dim = this._calculateCurrentDimensions(transformOptions),
```

and avoid adding the changes to calcRotationMatrix and calcTranslateMatrix and extend the api surface adding an option that is unclear when it should be used.

For `_calculateCurrentDimensions` i m sure we can find a better option, and i remember is said i would do. That does not bother me for now

Reply:
5deefc15